Tests/MongoSwiftTests/MongoCollection+BulkWriteTests.swift

```diff
@@ -0,0 +1,73 @@
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+final class MongoCollection_BulkWriteTests: XCTestCase {
+    static var allTests: [(String, (MongoCollection_BulkWriteTests) -> () throws -> Void)] {
+        return [
+            ("testEmptyRequests", testEmptyRequests),
+            ("testUnsupportedWriteModel", testUnsupportedWriteModel)
+        ]
+    }
+
+    static var client: MongoClient?
+    var coll: MongoCollection<Document>!
+
+    /// Set up the entire suite - run once before all tests
+    override class func setUp() {
+        super.setUp()
+        do {
+            client = try MongoClient()
+        } catch {
+            print("Setup failed: \(error)")
+        }
+    }
+
+    /// Set up a single test - run before each testX function
+    override func setUp() {
+        super.setUp()
+
+        guard let client = MongoCollection_BulkWriteTests.client else {
+            return XCTFail("Client is not initialized")
+        }
+
+        let collectionName = String(describing: self)
```

Comment:
Thoughts on using generated names for our test collections? This minimizes the chances that differing test files will use the same collection. In PHP, we actually have each test method use a unique collection name derived from its class and method names.

I don't see why we can't use a constant database name throughout the suite, unless certain tests end up requiring a specific name (e.g. documentation examples). Ideally, the database name could be picked up from a base test class.

Reply:
Created [SWIFT-206](https://jira.mongodb.org/browse/SWIFT-206) and [SWIFT-207](https://jira.mongodb.org/browse/SWIFT-207) to track that.